tenets/codelingo/k8/well-named-interface/codelingo.yaml

```diff
@@ -0,0 +1,30 @@
+funcs:
+  - name: beginsWith
+    type: asserter
+    body: |
+      function (start, str) {
+        return str.indexOf(start) == 0
+      }
+tenets:
+  - name: well-named-interface
+    flows:
+      codelingo/docs:
+        title: Well Named Interface
+        body: |
+          Please consider the package name when selecting an interface name, and avoid redundancy. e.g. `storage.Interface` is better than `storage.StorageInterface`.
+      codelingo/review:
+        comment: Interface name {{interfaceName}} has name of package {{packageName} redundantly prefixed.
+    query: |
+      import codelingo/ast/go
+            
+      go.file(depth = any):
+        go.ident:
+          name as packageName
+        go.decls:
+          go.gen_decl: # this can probably be made depth = any
```

Comment:
I think an interface is only going to be at the top level so no need for the comment. Although correct me if I'm wrong.

Reply:
You can define interfaces anywhere, we do need `depth = any`. https://play.golang.org/p/r-UyQhKljtQ

Reply:
Actually, I don't know for sure if the interface is always going to be at the top level. I imagine it's possible to have them defined elsewhere. Unsure about this.